### PR TITLE
Add ByteArray support for fs module

### LINKED ISSUE #35: Add methods for read/write ByteArrays
**Is your feature request related to a problem? Please describe.**
In my app I am handling several small binary files, which are not BASE64 encoded.

**Describe the solution you'd like**
I would like a readFileAsByteArray method which allows to read a file into a ByteArray and a vice versa write method.

### PR DESCRIPTION
**Pull request by svendroid**
Closes #35 

Make sure to merge https://github.com/suparngp/kotlin-multiplatform-projects/pull/34 before this.

### CODE REVIEW

**Comment by suparngp** [comment]
This one looks good overall. I ll quickly review it after merging the original PR for JVM.

File: fs/src/commonMain/kotlin/com/suparnatural/core/fs/FileSystem.kt [hunk 1]
```diff
@@ -67,6 +67,11 @@ expect object FileSystem {
      */
     fun readFile(pathComponent: PathComponent, encoding: ContentEncoding = ContentEncoding.Utf8): String?
 
+    /**
+     * Returns the contents of the file located at `path`as ByteArray.
+     */
+    fun readFileAsByteArray(path: String):ByteArray?
```
Comment (suparngp) [inline fs/src/commonMain/kotlin/com/suparnatural/core/fs/FileSystem.kt]: 1. Can you rename this to `readFile(path: String): ByteArray?` The return type is sufficient to mark that it reads as byte array.

2. Can you also add `readFile(path: PathComponent): ByteArray?` to keep it consistent with the rest of the API.

File: fs/src/commonMain/kotlin/com/suparnatural/core/fs/FileSystem.kt [hunk 2]
```diff
@@ -82,6 +87,12 @@ expect object FileSystem {
      */
     fun writeFile(pathComponent: PathComponent, contents: String, create: Boolean = false, encoding: ContentEncoding = ContentEncoding.Utf8): Boolean
 
+    /**
+     * Writes `contents` to the file located at `path`. If `create` is true, then file is created if it does not exist.
+     * Returns true if operation is successful, otherwise false.
+     */
+    fun writeFile(path: String, contents: ByteArray, create: Boolean): Boolean
```
Comment (suparngp) [inline fs/src/commonMain/kotlin/com/suparnatural/core/fs/FileSystem.kt]: Can you add `writeFile(path: PathComponent, contents: ByteArray, create: Boolean): Boolean` to keep it consistent with the rest of the API?

**Comment by svendroid** [comment]
Rebased and added the suggested changes.

**Pull request merged**